docs/src/main/paradox/events-by-tag.md

```diff
@@ -1,76 +1,52 @@
-The stream will initially do a query per time bucket to get up to the current 
-(many of these could be empty so very little over head, some may bring back many events). For example
-if you set the offset to be 30 days ago and a day time bucket, it will require 30 * nr queries to get each stream up to date. 
-If a particular partition contains many events then it is paged back from Cassandra based on demand.
-
-When no offset is provided EventsByTag queries start at the configured `first-time-bucket`. For new applications
-increase this to the day your application is initially deployed to speed up `NoOffset` queries.
-
-## Bucket sizes 
-
-Events by tag partitions are grouped into buckets to avoid a single partition for a tag
-getting to large. 
+## Consistency
 
-* 1000s of events per day per tag -- DAY
-* 100,000s  of events per day per tag -- HOUR
-* 1,000,000s of events per day per tag -- MINUTE
- 
-This setting can not be changed after data has been written.
- 
-More billion per day per tag? You probably need a specialised schema rather than a general library like this.
+Event by tag queries are eventually consistent, i.e. if a persist for an event has completed
+is does not guarantee that it will be visible in an events by tag query, but it will eventually.
 
-If a cassandra partition gets too large it will perform badly (and has a hard limit of 2 billion cells, a cell being similar to a column). 
-Given there is ~10 columns in the table there was a hard limit previously of 200million but performance would degrade before reaching this. 
+Order of the events is maintained per persistence id per tag by maintaining a sequence number for tagged events per persistence id.
+This is in addition to the normal sequence nr kept per persistence id.
 
-The support for `Minute` bucket size means that there are 1440 partitions per day. 
-Given a large C* cluster this should work until you have in the order of 10s of millions per minute
-when the write rate to that partition would likely become the bottleneck rather than partition size.
+Events from different persistent IDs are ordered by a [type 2 UUID a.k.a TimeUUID](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_2_(date-time_and_MAC_address,_DCE_security_version)).
+As events come from multiple nodes to deliver events in the correct order there is an `eventual-consistency-delay` to give
+events time to arrive in Cassandra and to the query in their final order. Without this events could be missed due to:
 
-The increased scalability comes at the cost of having to query more partitions on the read side so don't 
-use `Minute` unless necessary.
+* The repeated query to Cassandra is updated with the latest TimeUUID offset, meaning delayed events with an older TimeUUID won't be picked up
+* The query is restarted with the latest offset but there were delayed events time lower TimeUUIDs that hadn't been delivered yet
```

Comment:
Maybe this shouldn't be a list of reasons for missed events? Maybe just explain the first problem first, including the seqNr detection and backtracking solution.

Then the second as a separate problem, which we don't have an automatic solution for, or does the initial scanning help mitigating that risk?